vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -102,7 +102,12 @@ def _filter_interaction(
         try:
             custom_data_columns = model_manager[source_graph_id]["custom_data"]
         except KeyError as exc:
-            raise KeyError(f"No `custom_data` argument found for source graph with id {source_graph_id}.") from exc
+            raise KeyError(
+                f"Missing 'custom_data' for the source graph with id {source_graph_id}. "
```

Comment:
And maybe something similar as my suggestion above?

I overall feel we need to make clear three things:
1. have a `custom_data` argument
2. pass the input to the underlying plotly function
3. fill it in when configuring the graph

Maybe my suggestion helps, or a combination of the two?

Reply:
I updated the PR description with the latest changes. Do they look ok to you?